src/transactions/BumpSequenceOpFrame.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright 2017 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/BumpSequenceOpFrame.h"
+#include "transactions/TransactionFrame.h"
+#include "crypto/SignerKey.h"
+#include "database/Database.h"
+#include "main/Application.h"
+#include "medida/meter.h"
+#include "medida/metrics_registry.h"
+
+namespace stellar
+{
+using xdr::operator==;
+
+
+BumpSequenceOpFrame::BumpSequenceOpFrame(Operation const& op, OperationResult& res,
+                                     TransactionFrame& parentTx)
+    : OperationFrame(op, res, parentTx)
+    , mBumpSequence(mOperation.body.bumpSequenceOp())
+{
+}
+
+ThresholdLevel
+BumpSequenceOpFrame::getThresholdLevel() const
+{
+    // bumping sequence is low threshold
+    return ThresholdLevel::LOW;
+}
+
+bool
+BumpSequenceOpFrame::doApply(Application& app, LedgerDelta& delta,
+                           LedgerManager& ledgerManager)
+{
+    Database& db = ledgerManager.getDatabase();
+    // Account must be loaded from loadAccount to ensure data is fresh & account not deleted
+    AccountFrame::pointer bumpAccount =
```

Comment:
This is not needed. Similar code in other operations are only for protocol version < 8, and this op will fail in checkValid for protocol version < 9

mSourceAccount is set in following callstack:

OperationFrame::apply
OperationFrame::checkValid
OperationFrame::loadAccount
TransactionFrame::loadAccount <- this one does (very bad) caching for protocol version < 8

Reply:
Can you resolve with @bartekn on if this is sufficient or not?